app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt

```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
 
-    profileRepositoryFirestore.updateProfile(testUser, onSuccess = {}, onFailure = {})
+    profileRepositoryFirestore.addProfileToDatabase(testUser, onSuccess = {}, onFailure = {})
 
     verify(mockDocumentReference).set(testUser)
   }
 
   @Test
-  fun addProfileToDatabase_shouldCallFirestoreToAddUserProfile() {
-    val mockTaskVoid: Task<Void> = Tasks.forResult(null)
-    `when`(mockDocumentReference.set(testUser)).thenReturn(mockTaskVoid)
+  fun addActivity_shouldCallFirestoreToAddActivity() {
+    `when`(mockDocumentReference.update(eq("activities"), any())).thenReturn(Tasks.forResult(null))
 
-    profileRepositoryFirestore.addProfileToDatabase(testUser, onSuccess = {}, onFailure = {})
+    profileRepositoryFirestore.addActivity("1", "activityId", onSuccess = {}, onFailure = {})
 
-    verify(mockDocumentReference).set(testUser)
+    verify(mockDocumentReference).update(eq("activities"), any())
+  }
+
+  @Test
+  fun addLikedActivity_shouldCallFirestoreToAddLikedActivity() {
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forResult(null))
+
+    profileRepositoryFirestore.addLikedActivity("1", "activityId", onSuccess = {}, onFailure = {})
+
+    verify(mockDocumentReference).update(eq("likedActivities"), any())
+  }
+
+  @Test
+  fun removeLikedActivity_shouldCallFirestoreToRemoveLikedActivity() {
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forResult(null))
+
+    profileRepositoryFirestore.removeLikedActivity(
+        "1", "activityId", onSuccess = {}, onFailure = {})
+
+    verify(mockDocumentReference).update(eq("likedActivities"), any())
+  }
+
+  @Test
+  fun getUser_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.get()).thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.getUser(
+        "1",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
+  @Test
+  fun addActivity_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.update(eq("activities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.addActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
+  @Test
+  fun addLikedActivity_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.addLikedActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
```

Comment:
Handles failures in removing a liked activity.
Confirms the failure callback execution.